qiskit/aqua/components/optimizers/gsls.py

```diff
@@ -0,0 +1,325 @@
+
+    def get_support_level(self) -> Dict[str, int]:
+        """Return support level dictionary.
+
+        Returns:
+            A dictionary containing the support levels for different options.
+        """
+        return {
+            'gradient': Optimizer.SupportLevel.ignored,
+            'bounds': Optimizer.SupportLevel.supported,
+            'initial_point': Optimizer.SupportLevel.required
+        }
+
+    def optimize(self, num_vars: int,
+                 objective_function: callable,
+                 gradient_function: Optional[callable] = None,
+                 variable_bounds: Optional[List[Tuple[float, float]]] = None,
+                 initial_point: Optional[np.ndarray] = None) -> Tuple[np.ndarray, float, int]:
+        super().optimize(num_vars, objective_function, gradient_function,
+                         variable_bounds, initial_point)
+
+        if initial_point is None:
+            initial_point = np.random.normal(size=num_vars)
+        else:
+            initial_point = np.array(initial_point)
+
+        if variable_bounds is None:
+            var_lb = np.array([-np.inf] * num_vars)
+            var_ub = np.array([np.inf] * num_vars)
+        else:
+            var_lb = np.array([l for (l, _) in variable_bounds])
+            var_ub = np.array([u for (_, u) in variable_bounds])
+
+        x, x_value, n_evals, _ = self.ls_optimize(
+            num_vars, objective_function, initial_point, var_lb, var_ub
+        )
+
+        return x, x_value, n_evals
+
+    def ls_optimize(self, n: int, obj_fun: callable, initial_point: np.ndarray, var_lb: np.ndarray,
+                    var_ub: np.ndarray) -> Tuple[np.ndarray, float, int, float]:
+        """Run the line search optimization.
+
+        Args:
+            n: Dimension of the problem.
+            obj_fun: Objective function.
+            initial_point: Initial point.
+            var_lb: Vector of lower bounds on the decision variables. Vector elements can be -np.inf
+                    if the corresponding variable is unbounded from below.
+            var_ub: Vector of upper bounds on the decision variables. Vector elements can be np.inf
+                    if the corresponding variable is unbounded from below.
+
+        Returns:
+            Final iterate as a vector, corresponding objective function value,
+            number of evaluations, and norm of the gradient estimate.
+
+        Raises:
+            ValueError: If the number of dimensions mismatches the size of the initial point or
+                the length of the lower or upper bound.
+        """
+        if len(initial_point) != n:
+            raise ValueError('Size of the initial point mismatches the number of dimensions.')
+        if len(var_lb) != n:
+            raise ValueError('Length of the lower bound mismatches the number of dimensions.')
+        if len(var_ub) != n:
+            raise ValueError('Length of the lower bound mismatches the number of dimensions.')
```

Comment:
Shouldn't the error message here be 'Length of the **upper** bound mismatches....